spec/09-lexical-structure.md

```diff
@@ -678,7 +678,7 @@ The variable substitution accepts the following syntax:
 * [*name*](#names)
 * [*variable-name*](#names)
 
-*expression* works the same way as in [variable name creation operator](10-expressions.md#variable-name-creation-operator).
+The `${ expression }` syntax is deprecated in Hack. The expression must be such that if the enclosing braces were removed and the `$` prepended, then the resulting expression would be a legal <i>`string-variable`</i>. For example, `"${y[123]}"` is legal because `$y[123]` is a legal <i>`string-variable`</i>. No whitespace or comments are allowed inside the braces.
```

Comment:
Clause 4 states, “This specification is written from the perspective of strict mode only. A conforming implementation may provide modes other than strict, but they are outside the scope of this specification.”
 
Once we established that we'd limit things to strict, I believe I didn’t make any other references to non-strict modes, so to be consistent, we probably should continue to say nothing about them. If we do say something here, then that opens up the sticky question of “what else could/should we say in other places about other things?".